server/src/main/java/org/elasticsearch/action/admin/indices/refresh/TransportShardRefreshAction.java

```diff
@@ -64,6 +65,11 @@ protected ReplicationResponse newResponseInstance(StreamInput in) throws IOExcep
         return new ReplicationResponse(in);
     }
 
+    @Override
+    protected ClusterBlockLevel globalBlockLevel() {
+        return ClusterBlockLevel.METADATA_READ;
+    }
```

Comment:
Surprised this wasn't checked already, today we just return a trivial success before state recovery:

```
{
  "_shards" : {
    "total" : 0,
    "successful" : 0,
    "failed" : 0
  }
}
```

We could leave this as-is ofc.